mlflow/gateway/providers/togetherai.py

```diff
@@ -0,0 +1,470 @@
+
+        if "logitbias" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail=("Invalid parameter in payload."
+                "Parameter logitbias is not supported for togetherai.")
+            )
+
+        if "seed" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail=("Invalid parameter in payload."
+                "Parameter seed is not suuported of togetherai.")
+            )
+
+        if "presence_penalty" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail=("Invalid parameter in payload."
+                "Parameter presence_penalty is not supported for togetherai.")
+            )
+
+
+        return payload
+
+    @classmethod
+    def completions_streaming_to_model(cls, payload, config):
+        # parameters for streaming completions are the same as the standard completions
+        return TogetherAIAdapter.completions_to_model(payload, config)
+
+    @classmethod
+    def model_to_chat(cls, resp, config):
+        # Example response (https://docs.together.ai/reference/chat-completions):
+        #{
+        #   "id": "8448080b880415ea-SJC",
+        #   "choices": [
+        #    {
+        #        "message": {
+        #           "role": "assistant",
+        #           "content": "example"
+        #         }
+        #     }
+        #   ],
+        #   "usage": {
+        #     "prompt_tokens": 31,
+        #     "completion_tokens": 455,
+        #     "total_tokens": 486
+        #   },
+        #   "created": 1705090115,
+        #   "model": "mistralai/Mixtral-8x7B-Instruct-v0.1",
+        #   "object": "chat.completion"
+        #}
+        return chat.ResponsePayload(
+            id=resp["id"],
+            object="chat.completion",
+            created=resp["created"],
+            model=config.model.name,
+            choices=[
+                chat.Choice(
+                    index=idx,
+                    message=chat.ResponseMessage(
+                        role="assistant",
+                        content=c["message"]["content"],
+                    ),
+                    finish_reason=None,
+                )
+                for idx, c in enumerate(resp["choices"])
+            ],
+            usage=chat.ChatUsage(
+                prompt_tokens=resp["usage"]["prompt_tokens"],
+                completion_tokens=resp["usage"]["completion_tokens"],
+                total_tokens=resp["usage"]["total_tokens"],
+            ),
+        )
+
+    @classmethod
+    def model_to_chat_streaming(cls, resp, config):
+        #b'data: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051390,"choices":[{"index":0,"text":" According","logprobs":null,"finish_reason":null,"delta":{"token_id":6586,"content":" According"}}],"model":"mistralai/Mixtral-8x7B-v0.1","usage":null}\n\n'
```

Comment:
This is hardly readable, can we simplify the example or put reference link instead?

Reply:
I tried to find a reference link/response for ``TogetherAI`` stream response format. I couldn't find anything in their page, that's why I opted to run a query locally and copy paste the response.

Reply:
I updated it with a more readable format!

Reply:
Thanks a lot! It seem the formatter complains about the line length so you can break each sample to multiline, but it is now much easier to read!😄